MapboxNavigation/NavigationMapView.swift

```diff
@@ -1025,39 +1027,122 @@ extension Dictionary where Key == Int, Value: MGLStyleValue<NSNumber> {
+     - parameter mapView: the NavigationMapView.
+     - parameter identifier: the style identifier.
+     - parameter source: The Layer source provided.
+     - returns: an MGLStyleLayer that the map applies to the route.
+     */
     @objc optional func navigationMapView(_ mapView: NavigationMapView, routeCasingStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
+    /**
+     Tells the reciever that a route was selected.
+     - parameter mapView: the NavigationMapView.
+     - parameter route: The route that was selected.
+    */
     @objc(navigationMapView:didSelectRoute:)
     optional func navigationMapView(_ mapView: NavigationMapView, didSelect route: Route)
     
+    /**
+     Tells the reciever that a waypoint was selected.
+     - parameter mapView: the NavigationMapView.
+     - parameter waypoint: The waypoint that was selected.
+     */
     @objc(navigationMapView:didSelectWaypoint:)
     optional func navigationMapView(_ mapView: NavigationMapView, didSelect waypoint: Waypoint)
     
+    /**
+     Asks the reciever to return an MGLShape that describes the geometry of the route.
+     - note: If you implement this, you'll probably want to implement `simplifiedShapeDescribingRoute:` as well.
+     - parameter mapView: the NavigationMapView.
+     - parameter route: The route that the sender is asking about.
+     - returns: Optionally, a `MGLShape` that defines the shape of the route, or `nil` to use default behavior.
+     */
     @objc(navigationMapView:shapeDescribingRoute:)
     optional func navigationMapView(_ mapView: NavigationMapView, shapeDescribing route: Route) -> MGLShape?
     
+    /**
+     Asks the reciever to return an MGLShape that describes the geometry of the route at lower zoomlevels.
+     - note: If you implement `shapeDescribingRoute:`, you probably want to implement this too.
+     - parameter mapView: the NavigationMapView.
+     - parameter route: The route that the sender is asking about.
+     - returns: Optionally, a `MGLShape` that defines the shape of the route at lower zoomlevels, or `nil` to use default behavior.
+     */
     @objc(navigationMapView:simplifiedShapeDescribingRoute:)
     optional func navigationMapView(_ mapView: NavigationMapView, simplifiedShapeDescribing route: Route) -> MGLShape?
     
+    /**
+     Asks the reciever to return an MGLShape that describes the geometry of the waypoint.
+     - parameter mapView: the NavigationMapView.
+     - parameter waypoint: The waypoint that the sender is asking about.
+     - returns: Optionally, a `MGLShape` that defines the shape of the waypoint, or `nil` to use default behavior.
+     */
     @objc(navigationMapView:shapeDescribingWaypoints:)
     optional func navigationMapView(_ mapView: NavigationMapView, shapeFor waypoints: [Waypoint]) -> MGLShape?
     
+    /**
+     Asks the reciever to return a CGPoint to serve as the anchor for the User Icon.
```

Comment:
Nit: phrases such as “user icon” and “user location icon” don’t need to be title-cased.